test/openAPI/features/wallet_binding.feature

```diff
@@ -0,0 +1,31 @@
+@method=POST @endpoint=/wallet-binding
+Feature: The endpoint to validate wallet create wallet user id
+
+  @smoke @unit @positive
+  Scenario: Successfully validates the wallet and generates the wallet user id smoke type test
+    Given Wants to validate the wallet and generate wallet user id
+    When Send POST /wallet-binding request with given "n3fy2qkg9r7h2" as individualId and "OTP" as authFactorType and "encoded-json" as format and "string" as challenge and publicKey
+    Then Receive a response from the /wallet-binding endpoint
+    And The /wallet-binding response should be returned in a timely manner 15000ms
+    And The /wallet-binding endpoint response should have status 200
+    And The /wallet-binding endpoint response should match json schema with no errors
+
+  @negative
+  Scenario: Not able to generate the wallet binding because of unsupported challenge format
+    Given Wants to validate the wallet and generate wallet user id
+    When Send POST /wallet-binding request with given invalid format "n3fy2qkg9r7h2" as individualId and "OTP" as authFactorType and "invalid_format" as format and "string" as challenge and publicKey
+    Then Receive a response from the /wallet-binding endpoint
+    And The /wallet-binding response should be returned in a timely manner 15000ms
+    And The /wallet-binding endpoint response should have status 200
+    And The /wallet-binding endpoint response should match json schema with errors
+    And The /wallet-binding response should contain errorCode property equals to "unsupported_challenge_format"
+
+  @negative
+  Scenario: Not able to generate the wallet binding because of invalid public key
+    Given Wants to validate the wallet and generate wallet user id
+    When Send POST /wallet-binding request with given format "n3fy2qkg9r7h2" as individualId and "OTP" as authFactorType and "encoded-json" as format and "string" as challenge and invalid publicKey
+    Then Receive a response from the /wallet-binding endpoint
+    And The /wallet-binding response should be returned in a timely manner 15000ms
+    And The /wallet-binding endpoint response should have status 200
+    And The /wallet-binding endpoint response should match json schema with errors
+    And The /wallet-binding response should contain errorCode property equals to "invalid_public_key"
```

Comment:
Can we add one more test scenario with invalid auth callenge and expect "invalid_auth_challenge" as error code ?

Reply:
There is a problem with testing data inside arrays with mockoon. We can only check if arrays are empty or not. Let's keep it without this one for now.

Reply:
The comment is only about formation of the request with a invalid auth challange, nothing to check in array. Sorry may be I am not following you response correctly.

Reply:
@mdulko wanted to write that we have issue to mock data for this case, but it can be done ;)
I just need to know the list of valid auth challenge because for now in the spec is only "string". If there is no such list is it mean when in the request `challenge` field is empty or null should we received "invalid_auth_challenge" as error code?
@vishwa-vyom

Reply:
The challenge field should have the valid value for the authFactorType, for example if the authFactorType is OTP, the challenge field should contain actual OTP that is received in email. For our testing we can send a empty value and expect  "invalid_auth_challenge" as error code or we make this invalid challenge value configurable and get it implementation specific for each authFactoryType supported by the IDBB implementation.